### PR TITLE
Make `getCalInfoFromDB.py` install the correct `VREF_ADC` values

### PR DESCRIPTION
**Pull request by lpetre-ulb**
## Description
[Requires cms-gem-daq-project/gem-plotting-tools#264.]

This PR allows `getCalInfoFromDB.py` to install the `VREF_ADC` values on the CTP7 after a query to the database.

Once merged, the following procedure should allow a seamless operation at P5 regarding the DAC configuration:

1. Launch `testConnectivity.py` to establish the communication with the frontend (flags `--skipDACScan` and `--skipScurve`).
 2. Update all the configuration files with the information from the DB with `getCalInfoFromDB.py` (flags `--write2CTP7` and `--write2File`).
3. Restart `testConnectivity.py` where it was stopped with the flag `-f5`.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context

DAC scan failures encountered during commissioning at P5.

## How Has This Been Tested?

`VREF_ADC` values are correctly written to files in the `$GEM_PATH` and on the CTP7.

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: getCalInfoFromDB.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
     parser.add_argument("link",type=int,help="OH on AMC slot")
     parser.add_argument("-d","--debug",action="store_true",help="Prints additional information")
     parser.add_argument("--write2File",action="store_true",help="If Provided data will be written to appropriate calibration files")
-    parser.add_argument("--write2CTP7",action="store_true",help="If Provided IREF data will be sent to the VFAT3 config files on the CTP7")
+    parser.add_argument("--write2CTP7",action="store_true",help="If Provided VREF_ADC and IREF data will be sent to the VFAT3 config files on the CTP7")
```
Comment (jsturdy) [inline getCalInfoFromDB.py]: as we established, this message was misleading, as the retrieved data was only sent *if* `write2File` was specified.
in your current change, `write2CTP7` now *implies* `write2File`

**Review by jsturdy — DISMISSED** [review]

File: getCalInfoFromDB.py (hunk 1 above)
Comment (lpetre-ulb) [reply getCalInfoFromDB.py]: Right, I had forgotten about it. I'm so used to this kind of quirk...

Anyway, I added a commit which enables `--write2File` whenever `--write2CTP7` is set. This way, there is no more risk of user misuse.

File: getCalInfoFromDB.py [hunk 2]
```diff
@@ -8,9 +8,14 @@
     parser.add_argument("link",type=int,help="OH on AMC slot")
     parser.add_argument("-d","--debug",action="store_true",help="Prints additional information")
     parser.add_argument("--write2File",action="store_true",help="If Provided data will be written to appropriate calibration files")
-    parser.add_argument("--write2CTP7",action="store_true",help="If Provided VREF_ADC and IREF data will be sent to the VFAT3 config files on the CTP7")
+    parser.add_argument("--write2CTP7",action="store_true",help="If Provided VREF_ADC and IREF data will be sent to the VFAT3 config files on the CTP7 (implies --write2File)")
     args = parser.parse_args()
 
+    # Update the files on the DAQ machine whenever an update on the CTP7 is requested
+    # This avoids sending stall data on the CTP7 if --write2File is not set
```
Comment (jsturdy) [inline getCalInfoFromDB.py]: stall -> stale

**Review by jsturdy — COMMENTED** [review]
small typo, otherwise :+1:

**Comment by lpetre-ulb** [comment]
> small typo, otherwise +1

Fixed!

**Review by AndrewLevin — APPROVED** [review]

**Review by jsturdy — APPROVED** [review]

**Pull request merged**